src/StudioComponents/Loading.lua

```diff
@@ -37,32 +37,39 @@ local clock = os.clock
 local pi4 = 12.566370614359172 --4*pi
 
 return function(props: LoadingProperties): Frame
-    local isEnabled = getState(props.Enabled, true)
+	local isEnabled = getState(props.Enabled, true)
 	local time = Value(0)
 
+	local animThread = nil
+
 	local function startMotion()
-		local startTime = clock()
-		while unwrap(isEnabled) do
-			time:set(clock()-startTime)
-			task.wait(1/25) -- Springs will smooth out the motion so we needn't bother with high refresh rate here
+		if not unwrap(isEnabled) then return end
+
+		if animThread then
+			task.cancel(animThread)
+			animThread = nil
 		end
+
+		animThread = task.defer(function()
+			local startTime = clock()
+			while unwrap(isEnabled) do
+				time:set(clock()-startTime)
+				task.wait(1/25) -- Springs will smooth out the motion so we needn't bother with high refresh rate here
+			end
+		end)
 	end
 
-    local animThread = task.defer(startMotion)
-	Observer(isEnabled):onChange(function()
```

Comment:
No clue if you saw it, but the `:onChange()` connection to the Observer is never cleanup us as well. Once that's fixed, I think I can merge it.